chess/backend/src/game/logic.py

```diff
@@ -60,15 +82,29 @@ def _move_to_board_location(move: Move) -> Tuple[int]:
     to_coord = move.cord1.y * BOARD_WIDTH + move.cord1.x
     return (from_coord, to_coord)
 
+
 def make_move(request: WSGIRequest) -> JsonResponse:
     global global_board
     from_coord, to_coord = _get_coords_from_wsgi_request(request)
 
-    player_move = Move(newMove(from_coord, to_coord))
+    player_move = _get_move(from_coord, to_coord)
+
+    _check_for_castle(player_move)
 
     global_board = global_board.move(player_move)
     return _get_opponent_move()
 
+
+def _get_move(from_coord, to_coord) -> Move:
+    move = Move(newMove(from_coord, to_coord))
+    move = Move(move.cord0, move.cord1, global_board)
+    return move
```

Comment:
So is this the source of the bullshit. When you create a `Move` like on line 99, it doesn't do a lot of logic behind whether or not the move is a castle, en passant, etc. To get it to do that logic, you have to recreate the move by giving it a from_coord, to_coord, and the board. Then it will do extra work in figuring out if it is a castle move or not.